2024/half-2/day-12/array-filter/array-filter.test.ts

```
@@ -0,0 +1,71 @@
+    expect([1, 2, 3, 4].myFilter(isThisProductEven, 9)).toStrictEqual([2, 4]);
+    expect([1, 2, 3, 4].myFilter(isThisProductEvenArrowFn)).toStrictEqual([]);
+    expect([1, 2, 3, 4].myFilter(isThisProductEvenArrowFn, 10)).toStrictEqual(
+      []
+    );
+    expect([1, 2, 3, 4].myFilter(isThisProductEvenArrowFn, 9)).toStrictEqual(
+      []
+    );
+  });
+
+  test("sparse arrays", () => {
+    expect([, , ,].myFilter(isEven)).toStrictEqual([]);
```

Biome

[error] This array contains an empty slot.

Unsafe fix: Replace hole with undefined


(lint/suspicious/noSparseArray)